micromanager_gui/_gui_objects/_channel_widget.py

```diff
@@ -0,0 +1,94 @@
+
+
+class ChannelWidget(QWidget):
+    """Channel selector widget.
+
+    Parameters
+    ----------
+    channel_group : Optional[str]
+        Name of the group defining the microscope channels, by default will be guessed
+        using `mmc.getOrGuessChannelGroup`, and a dialog will be presented if there are
+        multiples
+    parent : Optional[QWidget]
+        Optional parent widget, by default None
+    """
+
+    def __init__(
+        self,
+        channel_group: Optional[str] = None,
+        parent: Optional[QWidget] = None,
+        *,
+        mmcore: Optional[CMMCorePlus] = None,
+    ):
+
+        super().__init__(parent)
+        self._mmc = mmcore or _core.get_core_singleton()
+
+        self._channel_group = channel_group or self._get_channel_group()
+
+        self.channel_wdg = self._create_channel_widget(self._channel_group)
+
+        self.setLayout(QVBoxLayout())
+        self.layout().setContentsMargins(0, 0, 0, 0)
+        self.layout().setSpacing(0)
+        self.layout().addWidget(self.channel_wdg)
+
+        self._mmc.events.systemConfigurationLoaded.connect(self._on_sys_cfg_loaded)
+        self._mmc.events.channelGroupChanged.connect(self._on_channel_group_changed)
+        self.destroyed.connect(self._disconnect_from_core)
+        self._on_sys_cfg_loaded()
+
+    def _get_channel_group(self) -> Optional[str]:
+        candidates = self._mmc.getOrGuessChannelGroup()
+        if len(candidates) == 1:
+            return candidates[0]
+        elif candidates:
+            dialog = ComboMessageBox(candidates, "Select Channel Group:", self)
+            if dialog.exec_() == dialog.DialogCode.Accepted:
+                return dialog.currentText()
+        return None
+
+    def _create_channel_widget(
+        self, channel_group: str
+    ) -> Union[PresetsWidget, QComboBox]:
+        if channel_group:
+            channel_wdg = PresetsWidget(channel_group)
+        else:
+            channel_wdg = QComboBox()
+            channel_wdg.setEnabled(False)
+        return channel_wdg
+
+    def _on_sys_cfg_loaded(self):
+        channel_group = self._channel_group or self._get_channel_group()
+        if channel_group is not None:
+            self._mmc.setChannelGroup(channel_group)
+
+    def _on_channel_group_changed(self, new_channel_group: str):
+        """When Channel group is changed, recreate combo."""
+        self._clear_previous_device_widget()
+        self._update_widget(new_channel_group)
+
+    def _clear_previous_device_widget(self):
+        self.channel_wdg.setParent(None)
+        self.channel_wdg.deleteLater()
+
+    def _update_widget(self, channel_group):
+        self.channel_wdg = self._create_channel_widget(channel_group)
+        self.layout().addWidget(self.channel_wdg)
+        if isinstance(
+            self.channel_wdg, PresetsWidget
+        ) and not self._mmc.getCurrentConfig(channel_group):
+            self.channel_wdg._combo.setStyleSheet("color: magenta;")
```

Comment:
```suggestion
```

Just let the PresetWidget handle this. I think this is incorrectly causing it to be magenta on creation.

Reply:
@ianhi you were right. To better handle that I made a small change in the PresetsWidget (#163).